pom.xml

```diff
@@ -120,6 +121,34 @@
     </profiles>
 
     <dependencies>
+        <!--
```

Comment:
Having dependencies to different OSGi versions seems wrong. Just upgrade to the latest parent which contains managed versions of those dependencies!

Reply:
Great suggestion.  Fixing it now.

Reply:
It seems serlvet-api dependemcy had been removed from Sling parent 26+.  What's the recommendation for Sling project using Servlet in general?  Should the version be maintained in module pom.xml?

Reply:
Upgrade to Sling Parent 29 for now as migrating away from Felix SCR annotations seems need a bit more works.  Probably worth another issue itself.   However, open to expanding the work into this issue if preferred.

Reply:
I think we should not mix Felix SCR with OSGi annotations in the same bundle, so it would be really appreciated if you could upgrade to parent 35.

Reply:
The change will require to upgrade a few more files that use the legacy R4 annotations plus migrating away from aven-scr-plugin.  Not sure how stable the the change would be but will give it a try.  Stay tune.

Reply:
@kwin - Sling JUnit Core uses <Embed-Packages/> to embed a few JUnit jars into the bundle.  Sling parent 35 has drop the support for maven-bundle-plugin.  We may need to stick with 34 until there a bnd.bnd that can simulate <Embed-Packages/> Felix Maven bundle plugin with in bnd-maven-plugin for Sling JUnit Core.

Thoughts?

Reply:
Instead of embedding you can either use https://bnd.bndtools.org/instructions/includeresource.html or rather https://bnd.bndtools.org/instructions/conditionalpackage.html with the bnd-maven-plugin.

Reply:
Thanks for the information.  Is there a Sling project with a bnd.bnd using "package coordinate in Maven repository" with -includresource: or -conditionalpackage:?  Just want to make sure the change uses an approach consistent with other Sling projects.

Reply:
@kwin - upgraded to Sling 35.  Can't find a bnd.bnd to refer to so I am taking a first stab at this.   Given the amount of changes involved, -includeresource: is used to preserve the original embedded packages flavor.